inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/KBUtility.java

```diff
@@ -0,0 +1,69 @@
+package de.tudarmstadt.ukp.inception.search.index.mtas;
+
+import java.util.Optional;
+
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.query.QueryEvaluationException;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.eclipse.rdf4j.repository.RepositoryResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.clarin.webanno.support.ApplicationContextProvider;
+import de.tudarmstadt.ukp.inception.kb.KnowledgeBaseService;
+import de.tudarmstadt.ukp.inception.kb.graph.KBConcept;
+import de.tudarmstadt.ukp.inception.kb.graph.KBInstance;
+import de.tudarmstadt.ukp.inception.kb.graph.KBObject;
+import de.tudarmstadt.ukp.inception.kb.graph.RdfUtils;
+import de.tudarmstadt.ukp.inception.kb.model.KnowledgeBase;
+
+public class KBUtility {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+
+    private KnowledgeBaseService kbService;
+
+    public KBUtility() {
+        super();
+        // TODO Auto-generated constructor stub
+        kbService = ApplicationContextProvider.getApplicationContext()
+                .getBean(KnowledgeBaseService.class);
+
+    }
+
+    public Optional<KBObject> readKBEntry(Project aProject, String aIdentifier)
+    {
+        for (KnowledgeBase kb : kbService.getKnowledgeBases(aProject)) {
+            try {
+                RepositoryConnection conn = kbService.getConnection(kb);
```

Comment:
Use try with resources; you do not close the connection right now. See the KBService

Reply:
@Rentier Can you refer me to an instance in KBService, where the connection is closed. I will close the connection as : 
if (conn.isOpen())
                    conn.close();

Reply:
```
try (RepositoryConnection conn = getConnection(kb)) {

}
```

Use try-with-resources, if you close by hand at the end then it can still be leaked if an exception is thrown before.